sdks/java/io/rrio/src/main/java/org/apache/beam/io/requestresponse/RedisClient.java

```diff
@@ -0,0 +1,183 @@
+ * RedisClient} is {@link java.io.Serializable}.
+ */
+class RedisClient implements SetupTeardown {
+
+  private final URI uri;
+
+  private transient @MonotonicNonNull JedisPooled jedis;
+
+  /**
+   * Instantiates a {@link RedisClient}. {@link URI} expected of the form: {@code
+   * redis://<host>:<port>}.
+   */
+  RedisClient(URI uri) {
+    this.uri = uri;
+  }
+
+  /**
+   * Decrement a value stored by the key, returning the resulting decremented value. Per Redis
+   * convention, sets the value to -1 for keys that do not exist. Naming of this method preserves
+   * that of the underlying {@link JedisPooled} client and performs a null check prior to execution.
+   */
+  long decr(String key) throws UserCodeExecutionException {
+    try {
+      return getSafeClient().decr(key);
+    } catch (JedisException e) {
+      throw new UserCodeExecutionException(e);
+    }
+  }
+
+  /**
+   * Get the long value stored by the key. Yields zero when key does not exist, keeping consistency
+   * with Redis convention. Consider using {@link #exists} to query key existance.
+   */
+  long getLong(String key) throws UserCodeExecutionException {
+    try {
+      return getSafeClient().decrBy(key, 0L);
+    } catch (JedisException e) {
+      throw new UserCodeExecutionException(e);
+    }
+  }
+
+  /** Query whether the key exists. */
+  boolean exists(String key) throws UserCodeExecutionException {
+    try {
+      return getSafeClient().exists(key);
+    } catch (JedisException e) {
+      throw new UserCodeExecutionException(e);
+    }
+  }
+
+  /**
+   * Increment a value stored by the key, returning the resulting decremented value. Sets the value
+   * to 1, key does not exist, per Redis convention. Naming of this method preserves that of the
+   * underlying {@link JedisPooled} client and performs a null check prior to execution.
+   */
+  long incr(String key) throws UserCodeExecutionException {
+    try {
+      return getSafeClient().incr(key);
+    } catch (JedisException e) {
+      throw new UserCodeExecutionException(e);
+    }
+  }
+
+  /**
+   * Query the size of a list identified by the key. Returns 0 if key does not exist, per Redis
+   * convention. Naming of this method preserves that of the underlying {@link JedisPooled} client
+   * and performs a null check prior to execution.
+   */
+  long llen(String key) throws UserCodeExecutionException {
+    try {
+      return getSafeClient().llen(key);
+    } catch (JedisException e) {
+      throw new UserCodeExecutionException(e);
+    }
+  }
+
+  /**
+   * Pushes items to the back ('right') of the list. Naming of this method preserves that of the
+   * underlying {@link JedisPooled} client and performs a null check prior to execution.
+   */
+  void rpush(String key, byte[]... items) throws UserCodeExecutionException {
+    try {
+      getSafeClient().lpush(key.getBytes(StandardCharsets.UTF_8), items);
```

Comment:
rpush or lpush? the delegated method disagrees, if this is right can you write a short comment explaining?

Reply:
Oh good catch!